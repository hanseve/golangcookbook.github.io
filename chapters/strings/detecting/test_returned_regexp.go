package main

import (
	"fmt"
	"regexp"
)

func main() {
	re := regexp.MustCompile("a.")
	fmt.Println(re.FindAllString("paranormal", -1))
	fmt.Println(re.FindAllString("paranormal", 2))
	fmt.Println(re.FindString("paranormal"))
	fmt.Println(re.FindAllString("graal", -1))
	fmt.Println(re.FindAllString("none", -1))

	re = regexp.MustCompile(`a(x*)b(y|z)c`)
	fmt.Printf("%q\n", re.FindStringSubmatch("-axxxbyc-"))
	fmt.Printf("%q\n", re.FindStringSubmatch("-abzc-"))
}
