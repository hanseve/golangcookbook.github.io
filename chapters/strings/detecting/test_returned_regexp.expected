[ar an al]
[ar an]
ar
[aa]
[]
["axxxbyc" "xxx" "y"]
["abzc" "" "z"]
